### PR TITLE
ENH: Add global nullable option

### LINKED ISSUE #50747: ENH: Add a global option to set use_nullable_dtypes to True if not explicitly specified
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

We added the ``use_nullable_dtypes`` to lots of new functions and methods for 2.0 (https://pandas.pydata.org/docs/dev/whatsnew/v2.0.0.html#configuration-option-mode-dtype-backend-to-return-pyarrow-backed-dtypes) We should add an option to set this to True if not explicitly given by the user to allow an easier opt in. 

The option would default to False for the time being.

The option would only work for functions where the keyword is implemented for the time being.

### Feature Description

pd.options.mode.nullable_dtypes = True

### Alternative Solutions

Specifying this for every function where the keyword is implemented, not easy to test or use.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #50747 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -49,6 +49,16 @@ The ``use_nullable_dtypes`` keyword argument has been expanded to the following
...
 
+To simplify the global opt-in a new option ``nullable_dtypes`` was added that allows to set
+the keyword argument globally to ``True`` if not specified directly. The option can be enabled
+through:
+
+.. ipython:: python
+
+    pd.options.mode.nullable_dtypes = True
+
+The option will only work in context with the keyword ``use_nullable_dtypes``.
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:60]:
```suggestion
The option will only work for functions with the keyword ``use_nullable_dtypes``.
```

File: pandas/core/config_init.py [hunk 2]
```diff
@@ -560,6 +560,22 @@ def use_inf_as_na_cb(key) -> None:
         validator=is_one_of_factory(["pandas", "pyarrow"]),
     )
 
+
+nullable_dtypes_doc = """
+: bool
+    If nullable dtypes should be returned. This is only applicable to functions
+    where ``use_nullable_dtypes`` is implemented.
```
Comment (mroeschke) [inline pandas/core/config_init.py:567]:
```suggestion
    where the ``use_nullable_dtypes`` keyword is implemented.
```

**Comment by phofl** [comment]
Thx!

**Review by mroeschke — COMMENTED** [review]
Looks fairly good. Just a merge conflict

**Comment by phofl** [comment]
resolved

**Pull request merged**